scene/gui/tab_bar.cpp

```diff
@@ -1266,7 +1266,7 @@ void TabBar::add_tab(const String &p_str, const Ref<Texture2D> &p_icon) {
 	queue_redraw();
 	update_minimum_size();
 
-	if (tabs.size() == 1) {
+	if (tabs.size() == 1 && !deselect_enabled) {
```

Comment:
```suggestion
	if (!deselect_enabled && tabs.size() == 1) {
```
Cheaper checks should be performed first.

Reply:
Right. Also fits the style of the source file better.